File: solutions/dart/strain/1/lib/strain.dart
// Vowed to Shree DR.MDD 💫

class Strain {
  List<T> keep<T>(List<T> input, bool test(T item)) {
    var filtered = <T>[];
    for (final val in input)
      if (test(val))
        filtered.add(val);
    return filtered;
  }

  List<T> discard<T>(List<T> input, bool test(T item)) {
    return keep(input, (val) => !test(val));
  }
}

File: solutions/julia/strain/1/strain.jl
# Dedicated to Shree DR.MDD

function keep(vals, testfn)
    chosen = []
    for item in vals
        testfn(item) && push!(chosen, item)
    end
    chosen
end

function discard(vals, testfn)
    dropped = []
    for item in vals
        !testfn(item) && push!(dropped, item)
    end
    dropped
end
